pkg/capabilities/utils.go

```diff
@@ -59,15 +61,19 @@ func SetResponse(response *CapabilityResponse, migrated bool, value proto.Messag
 // FromValueOrAny extracts the value from either a values.Value or an anypb.Any, returning true if the value was migrated to use pbany.Any.
 func FromValueOrAny(value values.Value, any *anypb.Any, into proto.Message) (bool, error) {
 	if any == nil {
+		// Check if the underlying concrete value is nil
+		if v, ok := value.(*values.Map); ok && v == nil {
```

Comment:
this isn't exhaustive, but covers the case that an empty config values map is passed in.